### PR TITLE
Add changelog for 10.0.0-rc.1

### PR DESCRIPTION
**Pull request by nishant-karajgikar**
This PR adds the changelog for `v10.0.0-rc.1`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Mapbox welcomes participation and contributions from everyone.
 
-## main
+## 10.0.0-rc.1 - June 9, 2021
```
Comment (neelmistry94) [inline CHANGELOG.md:5]: NIT: I think we should leave the ##main in the changelog for post release changes

**Comment by neelmistry94** [comment]
@nishant-karajgikar we should add a bug fix entry for https://github.com/mapbox/mapbox-maps-ios/pull/443.

**Review by macdrevx — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,15 +4,21 @@ Mapbox welcomes participation and contributions from everyone.
 
 ## main
 
+## 10.0.0-rc.1 - June 9, 2021
+
 ### Breaking changes ⚠️
```
Comment (knov) [inline CHANGELOG.md:9]:
```suggestion

**The Mapbox Maps SDK for Android has moved to release candidate status and is now ready for production use.**

### Breaking changes ⚠️
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,15 +4,30 @@ Mapbox welcomes participation and contributions from everyone.
...
+- Removed `CacheManager`. In the following releases, an API to control temporary map data may be provided. ([#440](https://github.com/mapbox/mapbox-maps-ios/pull/440))
+- Changed `ResourceOptions.cachePathURL` to `dataPathURL` and removed `cacheSize`. ([#440](https://github.com/mapbox/mapbox-maps-ios/pull/440))
 
 ### Features ✨ and improvements 🏁
-- Fixed an issue where location updates were only happening via `heading: CLHeading`. You may now specify `mapView.location.options.puckBearingSource = .course` so that location puck will update based on `course: CLDirection`. Note that this value will use `heading: CLHeading` by default. ([#428](https://github.com/mapbox/mapbox-maps-ios/pull/428))
+- Allows a developer to choose whether the puck is oriented based on `heading` or `course` via a new `puckBearingSource` option in `mapView.location.options`. By default, the puck will be oriented using `heading`. ([#428](https://github.com/mapbox/mapbox-maps-ios/pull/428))
+- Introduced `OfflineRegionManager.setOfflineMapboxTileCountLimit()` method. ([#440](https://github.com/mapbox/mapbox-maps-ios/pull/440))
```
Comment (knov) [inline CHANGELOG.md:19]:
```suggestion
```

**Pull request merged**